Source/FlowEditor/Private/DetailCustomizations/FlowNode_CustomNodeBaseDetails.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright https://github.com/MothCocoon/FlowGraph/graphs/contributors
+
+#include "DetailCustomizations/FlowNode_CustomNodeBaseDetails.h"
+#include "FlowAsset.h"
+#include "Nodes/Route/FlowNode_CustomNodeBase.h"
+
+#include "DetailCategoryBuilder.h"
+#include "DetailWidgetRow.h"
+#include "PropertyEditing.h"
+#include "Widgets/Input/SComboBox.h"
+#include "Widgets/Text/STextBlock.h"
+#include "Widgets/SWidget.h"
+
+
+void FFlowNode_CustomNodeBaseDetails::CustomizeDetails(IDetailLayoutBuilder& DetailLayout)
+{
+	// Subclasses must override this function (and call CustomizeDetailsInternal with the localized text)
+	checkNoEntry();
+}
+
+void FFlowNode_CustomNodeBaseDetails::CustomizeDetailsInternal(IDetailLayoutBuilder& DetailLayout, const FText& CustomRowNameText, const FText& EventNameText)
+{
+	DetailLayout.GetObjectsBeingCustomized(ObjectsBeingEdited);
+
+	if (ObjectsBeingEdited[0].IsValid())
+	{
+		UFlowNode_CustomNodeBase* FlowNodeBase = CastChecked<UFlowNode_CustomNodeBase>(ObjectsBeingEdited[0]);
+		CachedEventNameSelected = MakeShared<FName>(FlowNodeBase->GetEventName());
+	}
+
+	IDetailCategoryBuilder& Category = CreateDetailCategory(DetailLayout);
+
+	Category.AddCustomRow(CustomRowNameText)
+		.NameContent()
+			[
+				SNew(STextBlock)
+					.Text(EventNameText)
+			]
+		.ValueContent()
+			.HAlign(HAlign_Fill)
+			[
+				SAssignNew(EventTextListWidget, SComboBox<TSharedPtr<FName>>)
+					.OptionsSource(&EventNames)
+					.OnGenerateWidget(this, &FFlowNode_CustomNodeBaseDetails::GenerateEventWidget)
+					.OnComboBoxOpening(this, &FFlowNode_CustomNodeBaseDetails::OnComboBoxOpening)
+					.OnSelectionChanged(this, &FFlowNode_CustomNodeBaseDetails::PinSelectionChanged)
+					[
+						SNew(STextBlock)
+							.Text(this, &FFlowNode_CustomNodeBaseDetails::GetSelectedEventText)
+					]
+			];
+}
+
+void FFlowNode_CustomNodeBaseDetails::OnComboBoxOpening()
+{
+	RebuildEventNames();
+}
+
+void FFlowNode_CustomNodeBaseDetails::RebuildEventNames()
+{
+	EventNames.Empty();
+
+	check(CachedEventNameSelected.IsValid());
+	EventNames.Add(CachedEventNameSelected);
+
+	if (ObjectsBeingEdited[0].IsValid() && ObjectsBeingEdited[0].Get()->GetOuter())
+	{
+		const UFlowAsset* FlowAsset = CastChecked<UFlowAsset>(ObjectsBeingEdited[0].Get()->GetOuter());
+		TArray<FName> SortedNames = BuildEventNames(*FlowAsset);
+
+		if (bExcludeReferencedEvents)
```

Comment:
_The contents of this "if" were the primary difference between the CustomInput and CustomOutput versions of this function.  The unified version uses this bool to choose whether it should do the CustomInput version (true) or the CustomOutput version (false)._